apps/src/util/experiments.js

```diff
@@ -24,23 +24,42 @@ experiments.getQueryString_ = function () {
   return window.location.search;
 };
 
-experiments.getEnabledExperiments = function () {
+experiments.getStoredExperiments_ = function () {
   try {
     const jsonList = localStorage.getItem(STORAGE_KEY);
-    const enabled = jsonList ? JSON.parse(jsonList) : [];
-    return enabled;
+    const storedExperiments = jsonList ? JSON.parse(jsonList) : [];
+    const now = Date.now();
+    const enabledExperiments = storedExperiments.filter(
+        experiment => experiment.expiration > now);
+    if (enabledExperiments.length < storedExperiments.length) {
+      trySetLocalStorage(STORAGE_KEY, JSON.stringify(enabledExperiments));
+    }
+    return enabledExperiments;
   } catch (e) {
     return [];
   }
 };
 
+experiments.getEnabledExperiments = function () {
+  return this.getStoredExperiments_().map(experiment => experiment.key);
+};
+
 experiments.setEnabled = function (key, shouldEnable) {
-  const allEnabled = this.getEnabledExperiments();
-  if (allEnabled.indexOf(key) < 0 && shouldEnable) {
-    allEnabled.push(key);
-    trackEvent(GA_EVENT, 'enable', key);
-  } else if (allEnabled.indexOf(key) >= 0 && !shouldEnable) {
-    allEnabled.splice(allEnabled.indexOf(key), 1);
+  const allEnabled = this.getStoredExperiments_();
+  const experimentIndex =
+    allEnabled.findIndex(experiment => experiment.key === key);
```

Comment:
Does this get polyfilled by our build process?  https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex#Browser_compatibility